src/libsyntax/feature_gate.rs

```diff
@@ -1430,6 +1433,14 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
         visit::walk_expr(self, e);
     }
 
+    fn visit_arm(&mut self, arm: &'a ast::Arm) {
+        if arm.used_beginning_vert {
+            gate_feature_post!(&self, match_beginning_vert,
+                               arm.pats[0].span,
```

Comment:
Is there a better span to use here?

I suppose used_beginning_vert could become Option<Span>...

Reply:
It should be possible to call `emit_feature_err` immediately from `parse_arm`, the precise span is available there.

In addition, changes to AST (`used_beginning_vert: bool`) will likely break `rustfmt` and you'll have to send PR to rustfmt repo as well, do submodule updates, etc, so I recommend against it.

Reply:
Oh, I didn't know that features were already available during parsing!

Reply:
I don't know if they are available as well, but it's worth trying, because the alternative is not especially pleasant.